### PR TITLE
US-4408 Added non-proxied getWindowSize

### PR DESCRIPTION
**Pull request by simongranger**
Added a non-proxied version of getWindowSize as iOS native version was returning the virtual pixel width and height.

### CODE REVIEW

**Comment by pfoster-youitv** [comment]
have you tried it on iOS and Android?
if it's in the general.js, won't that mean the Android one is no longer proxied?

**Comment by simongranger** [comment]
Correct, it will apply to all platforms

File: lib/commands/general.js [hunk 1]
```diff
@@ -20,4 +22,27 @@ commands.getPageSource = async function () {
...
+  var commandJSON = JSON.stringify(commandObject);
+
+  let data = await this.executeSocketCommand(commandJSON);
+
+  let result;
+  try {
+    result = JSON.parse(data);
+  } catch (e) {
+    throw new Error("Bad response from window_size");
+  }
```
Comment (pfoster-youitv) [inline lib/commands/general.js]: Is this what will be thrown if someone uses this version of the appium driver (js) with an older version of the engine? (i.e. one that doesn't support this command?)

**Comment by simongranger** [comment]
It was tested on Android

File: lib/commands/general.js [hunk 2]
```diff
@@ -20,4 +22,30 @@ commands.getPageSource = async function () {
...
+
+  // get status returned and handle errors returned from server
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_NO_SUCH_WINDOW) {
+    throw new Error("Could not find the requested surface");
```
Comment (pfoster-youitv) [inline lib/commands/general.js]: add a period :bike:

**Pull request merged**